### PR TITLE
tr(PR template): add Github Pull request template for external contributions

### PR DESCRIPTION
**Pull request by educhastenier**

### CODE REVIEW

**Comment by danila-m** [comment]
Don't we want to add the relevant JIRA tickets in the commit messages when possible?

**Comment by educhastenier** [comment]
@danila-m external contributions never have JIRA associated

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Insert your contribution description here.
+# Ensure your contribution fills the Contributing Guidelines described here:
+#        https://github.com/bonitasoft/bonita-doc/blob/7.3/CONTRIBUTING.md
+# If applied, this commit will...
+(): 
+
+# Commit title MUST respect the following format: <type>(<scope>): <subject>
```
Comment (laurentleseigneur) [inline .github/PULL_REQUEST_TEMPLATE.md]: you can also talk about maximum size to avoid truncated title in PR

Comment (tbouffard) [reply .github/PULL_REQUEST_TEMPLATE.md]: 👍

Comment (educhastenier) [reply .github/PULL_REQUEST_TEMPLATE.md]: I'll do that

**Review by lcustodio — COMMENTED** [review]
Eventually too much text, can lead to user being afraid to modify something.
Possible alternatives are:
1. Move the description of Title and Description to the CONTRIBUTING article.
2. Add some text as markdown comments. (Not necessarily the best as verified with @julienmege)

**Comment by lcustodio** [comment]
@educhastenier can you please review the new format? What do you think?

**Comment by laurentleseigneur** [comment]
shouldn't we add here some info about licensing and explain that external contributor (out of Bonitasoft) should sign some king of virtual paper and agree that we finally keep ownership on his/her contributed data ?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<!-- 
+Insert your contribution description here.
+Ensure your contribution fills the Contributing Guidelines described here:
+https://github.com/bonitasoft/bonita-doc/blob/7.3/CONTRIBUTING.md
```
Comment (cpuy) [inline .github/PULL_REQUEST_TEMPLATE.md:4]: I guess we should point to another branch than `7.3` isnt it ?

**Comment by cpuy** [comment]
If you want my opinion, Pull request template ALWAYS bother me.
So I would put only the first comment to tell the user to read the guidelines.

Also I have the feeling that there is a mix between commit messages and pull request messages here

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (educhastenier) [reply .github/PULL_REQUEST_TEMPLATE.md:4]: This is the first version where we put the contributing guidelines, and they are the same on the branches above.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
+<!-- 
+Insert your contribution description here.
+Ensure your contribution fills the Contributing Guidelines described here:
+https://github.com/bonitasoft/bonita-doc/blob/7.3/CONTRIBUTING.md 
+-->
+
+<!-- 
+Special Attention to TITLE format
+https://github.com/bonitasoft/bonita-doc/blob/7.3/CONTRIBUTING.md#title
```
Comment (educhastenier) [inline .github/PULL_REQUEST_TEMPLATE.md:9]: Actually, title format is more for commits than for PR

Comment (educhastenier) [reply .github/PULL_REQUEST_TEMPLATE.md:9]: Is this link `#title` working? I think we should add a `<a id="title" />` for the link to work.

**Comment by educhastenier** [comment]
@cpuy yes, there is a mix between commit message format and PR template, that's why we're limiting the PR template to a minimum of text, and pointing to the guidelines where we can add as much description there as we want

**Comment by cpuy** [comment]
ok but as far as I know we don't have any guidelines for PR title and messages.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (lcustodio) [reply .github/PULL_REQUEST_TEMPLATE.md:9]: it works as soon the new contributing file is available. e.g.: https://github.com/bonitasoft/bonita-doc/blob/7.3/README.md#taxonomy

Comment (lcustodio) [reply .github/PULL_REQUEST_TEMPLATE.md:9]: `title format is more for commits than for PR`.... I do agree however, PR is the best way we have to give a hint about best practices.

**Comment by lcustodio** [comment]
@cpuy @educhastenier the commit guidelines are there to help during the creation of the PR request right? As soon we squash the changes the metadata of the commit is rewritten.

**Comment by cpuy** [comment]
hum.... IMO squash button is a bad practice. Committer should take care of making descriptive commit, not the reviewer.
Nevertheless since our goal is to have more external contribution, i guess having a well documented PR with poorly descriptive commits could do the job and reviewer take care of squashing commits when merging.
Is it the way you want to do ?

**Comment by lcustodio** [comment]
@laurentleseigneur I've talked about the licensing with @nchabanoles about it, he is going to check and come back to us.

**Comment by lcustodio** [comment]
@cpuy Exactly! We are on the same page about the committer responsibility to write a clear message. 

However, in the docs repo making sure that the PR is clear and well made will allow us to understand the changes while allowing more people to contribute.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (laurentleseigneur) [reply .github/PULL_REQUEST_TEMPLATE.md:9]: in that case we may suggest that PR should be squashed/contains only 1 commit

**Review by laurentleseigneur — APPROVED** [review]

**Comment by laurentleseigneur** [comment]
LGTM

**Comment by lcustodio** [comment]
Everything seems good, I'm going to squash and merge into 7.3

**Pull request merged**